scripts/utils.py

```diff
@@ -6,130 +6,199 @@
 import os
 from itertools import chain
 import gc
-
+import requests
+import editdistance as ed
+import xml.etree.ElementTree as ET
+import re 
+
+
+def convertToType(value):
+    try:
+        return int(value)
+    except Exception:
+        try:
+            return float(value)
+        except Exception:
+            return value
+
+def pandas_from_xml(data_file_name):
+  xml = ET.parse(data_file_name)
+  root = xml.getroot()
+  string = "\{(.*?)\}"
+  prefix = re.search(string,root.tag)[0]
+  data_dict = {}
+  for tname in root.findall('.//' + prefix + 'ClinicalVariables'):
+      attribute = tname.find(prefix + 'TypeName').text
+      if attribute not in data_dict.keys():
+          data_dict[attribute] = []
+      data_dict[attribute].append(convertToType(tname.find(prefix + 'Value').text))
+  data = pd.DataFrame.from_dict(data_dict)
+  del data_dict
+  return data
+
+def pandas_from_csv(data_file_name):
+  data = pd.read_csv(data_file_name)
+  return data
+
+def data_to_pandas(data_file_name):
+  data = None
+  if data_file_name.endswith(".xml"):
+      data=pandas_from_xml(data_file_name)
+  elif data_file_name.endswith(".csv"):
+      data=pandas_from_csv(data_file_name)
+  return data
 
 def sort_attributes(attribute_type_mapping):
     def sorting(type_):
         return len(attribute_type_mapping[type_])
     return sorting
 
+def lcs(X, Y): 
+  m = len(X) 
+  n = len(Y) 
+  T = np.zeros((n+1,m+1))
+  for j in range(1,m+1): 
+    for i in range(1,n+1): 
+      if Y[i-1] == X[j-1]: T[i,j] = T[i-1,j-1]+1
+      else: T[i,j] = max(T[i-1,j], T[i,j-1])      
+  return T[n,m] 
+
+def load_mesh_mapping_file(mesh_mapping_file_path) -> dict:
+  with open(mesh_mapping_file_path, 'r') as f:
+    return json.load(f)
+
+def map_mesh_to_values(mesh_code, loaded_mesh_mapping) -> list:
+  for term in loaded_mesh_mapping:
+    if term['code'] == mesh_code: return term['name']
+
+def map_values_to_mesh(value, url = "https://goldorak.hesge.ch:8082/transmesh/translate/"):
```

Comment:
URL should be an environment variable so we can change the API without changing the code.